components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java

```diff
@@ -440,6 +440,35 @@ public static IdentityRecoveryClientException handleClientException(IdentityReco
         return IdentityException.error(IdentityRecoveryClientException.class, error.getCode(), errorDescription, e);
     }
 
+    /**
+     * Handles client exceptions by creating an instance of {@link IdentityRecoveryClientException}
+     * with the specified error details.
+     * If the provided data is not blank, it is formatted into the error message; otherwise,
+     * the default error message is used.
+     *
+     * @param error   The predefined error message from {@link IdentityRecoveryConstants.ErrorMessages}.
+     * @param data    Additional data to be included in the error message (can be {@code null} or blank).
+     * @param message A brief message describing the error.
+     * @param e       The underlying cause of the exception.
+     * @return An instance of {@link IdentityRecoveryClientException} with the provided details.
+     * @throws IdentityRecoveryClientException If an error occurs while instantiating the exception.
+     */
+    public static IdentityRecoveryClientException handleClientException(IdentityRecoveryConstants.ErrorMessages error,
```

Comment:
Shall we change this to,

handleClientException(String errorCode, String message, String description, String data, Throwable e) ?

Reply:
Fixed c5d9dcc6186f13148911ace37c858f6e6c7782ec